webapp/channels/src/components/quick_switch_modal/quick_switch_modal.tsx

```diff
@@ -137,7 +147,7 @@ export default class QuickSwitchModal extends React.PureComponent<Props, State>
             }
             switchToChannel(selectedChannel).then((result: ActionResult) => {
                 if ('data' in result) {
-                    this.onHide();
+                    this.onSelect();
```

Comment:
A few lines below this, we still call `this.onHide()`, and I think that should also be changed to `this.onSelect()`